src/index.ts

```diff
@@ -0,0 +1,833 @@
+import { EventEmitter } from 'events'
+
+export const DEFAULT_HOST = '127.0.0.1'
+export const DEFAULT_PORT = 9003
+export const DEFAULT_PUBLIC_PORT = 9005
+export const DEFAULT_PRIVATE_PORT = 9006
+export const DEFAULT_REPLICAS = 3
+export const DEFAULT_ID = 0
+export const DEFAULT_TOPIC = 'message'
+export const DEFAULT_PARTITIONS = 1
+export const DEFAULT_REPLICATION_FACTOR = 1
+export const DEFAULT_IGNORE_RACK_ASSIGNMENT = false
+export const DEFAULT_MIN_ID = 0
+export const DEFAULT_OFFSET = 0
+export const DEFAULT_OFFSET_FROM = 'beginning'
+
+/**
+ * Top-level Fluvio Client options;
+ *
+ * These options are unstable and may be categorized into
+ * more specific interfaces; For now, this is the global options
+ * used by the entirety of the fluvio client;
+ *
+ * It is best practice to use explicit arguments directly with the class methods
+ * If an option or argument does not exist that is optional, a default value will be
+ * used.
+ */
+export interface Options {
+    host?: string
+    port?: number
+    ingress?: IngressAddr[]
+    rack?: string
+    encryption?: Encryption
+    minId?: number
+    replicas?: number
+    env?: EnvVar[]
+    replication?: ReplicationConfig
+    storage?: StorageConfig
+    partitions?: number
+    replicationFactor?: number
+    ignoreRackAssignment?: boolean
+    maps?: PartitionMap[]
+    topic?: string
+    partition?: number
+    id?: number
+    offsetIndex?: number
+    offsetFrom?: string
+}
+
+export interface TopicProducer {
+    sendRecord(data: String, partition: number): Promise<void>
+}
+
+/**
+ * # Topic Producer
+ *
+ * ## Overview
+ *
+ * An interface for producing events to a particular topic
+ *
+ * A TopicProducer allows you to send events to the specific topic it was initialized for.
+ * Once you have a TopicProducer, you can send events to the topic,
+ * choosing which partition each event should be delivered to.
+ *
+ * ## Example Construction
+ *
+ * Do not construct this manually, instead use the following method:
+ *
+ * ```TypeScript
+ * const fluvio = new Fluvio({ host, port })
+ *
+ * await fluvio.connect();
+ *
+ * const producer = await fluvio.topicProducer("topic-name")
+ * ```
+ *
+ * This class constructor is used internally by the Fluvio client
+ * to provision a topic producer.
+ *
+ * The `inner` object is a private object that is the native module
+ * created by the top-level Fluvio client.
+ *
+ * This class is not intended to be constructed manually.
+ *
+ */
+export class TopicProducer {
+    private inner: TopicProducer
+    /**
+     * This method is not intended to be used directly. This is a helper
+     * method used by the `Fluvio` class to pass in a native object,
+     * along with top-level Fluvio client options, if any.
+     *
+     * @param inner The native node module created by `new Fluvio().connect().topicProducer()`
+     * @param options Top-level Fluvio client options
+     */
+    constructor(inner: TopicProducer, options?: Partial<Options>) {
```

Comment:
commit b9e4c10 adds a static method and a private constructor for this; Now it will not allow a direct use of `new TopicProducer` and instead use a factory pattern for creating the instance. This is the same for PartitionConsumer.